scripts/lib/CIME/case/case.py

```diff
@@ -1487,6 +1509,20 @@ def get_mpirun_cmd(self, job=None, allow_unresolved_envvars=True, overrides=None
         if self.get_value("BATCH_SYSTEM") == "cobalt":
             mpi_arg_string += " : "
 
+        ngpus_per_node = self.get_value("NGPUS_PER_NODE")
+        if ngpus_per_node > 0:
+            # generate a wrapper script under the case dir to set the device id for each MPI rank
+            casedir = self.get_value("CASEROOT")
+            with open (casedir+'/set_device_rank.sh', 'w+') as f:
```

Comment:
This code seems like it may be pretty specific to casper.  Can we think of a way to do it that will be more adaptable to other systems?

Reply:
Thanks @jedwards4b . Could you provide more details about which part of this code do you think is too specific to Casper?

Reply:
I am referring to the set_device_rank.sh script and the way that it is included in the mpi_arg_string.  I think that it's okay as is for now, but we may need to revisit this as we extend gpu support to more systems.

Reply:
Thanks for the details, @jedwards4b. Now I see your point and I agree that this setting may be too specific. Since Casper is the only machine I have the access with GPU, I could not test this configuration in another machine. But I could include a comment section here in case we need to revisit it in the future.